CMakeLists.txt

```diff
@@ -67,86 +48,101 @@ set( WIN32_BOOTSTRAP_SOURCES
 
-function(_COMMON_PROJECT_OPTIONS)
-	include_directories(${INCLUDE_DIR} ${LIB_INCLUDE_DIR})
-	link_directories(${LIB_DIR}/boost/lib/)
-
-	add_compile_options(-std:c++latest)
+function(_COMMON_PROJECT_OPTIONS TARGET_NAME)
+	target_include_directories(${TARGET_NAME}
+		PRIVATE
+			sources/inc
+	)
+	target_link_libraries(${TARGET_NAME}
+		PRIVATE
+			Boost::boost
+			Boost::filesystem
+	)
+
+	if(MSVC)
+		add_compile_options(-std:c++latest)
+	endif(MSVC)
 
 	if (WARN_VERBOSE)
-	add_compile_options(-Wall)
+		add_compile_options(-Wall)
 	else()
-	add_compile_options(-W4 -WX)
+		add_compile_options(-W4 -WX)
 	endif(WARN_VERBOSE)
 endfunction()
 
 # ==============================================================================
 
-project(utility)
-
-_common_project_options()
 add_library(utility STATIC ${UTILITY_SOURCES})
+_common_project_options(utility)
 
 # ______________________________________________________________________________
 
-project(oglbase)
-
-_common_project_options()
-_find_gl_libraries(GL_LIBRARIES)
 add_library(oglbase STATIC ${OGLBASE_SOURCES})
-target_link_libraries( oglbase
-					   PUBLIC
-					   "${GL_LIBRARIES}"
-					   )
+target_link_libraries(oglbase
+	PUBLIC
+		OpenGL::GL
+		GLEW::GLEW
+)
+_common_project_options(oglbase)
 
 # ______________________________________________________________________________
 
-project(uibase)
-
-_common_project_options()
 add_library(uibase STATIC ${UIBASE_SOURCES})
-target_link_libraries( uibase
-					   PUBLIC
-					   imgui
-					   PRIVATE
-					   oglbase
-					   )
+target_link_libraries(uibase
+	PUBLIC
+		imgui::imgui
+	PRIVATE
+		oglbase
+)
+_common_project_options(uibase)
 
 # ______________________________________________________________________________
 
-project(shaderunner)
-
-_common_project_options()
 add_library(shaderunner STATIC ${SHADERUNNER_SOURCES})
-target_link_libraries( shaderunner
-					   PRIVATE
-					   utility
-					   oglbase
-					   )
+target_link_libraries(shaderunner
+	PRIVATE
+		utility
+		oglbase
+)
+_common_project_options(shaderunner)
 
 # ______________________________________________________________________________
 
-project(win32_bootstrap)
-
-_common_project_options()
 add_executable(win32_bootstrap WIN32 ${WIN32_BOOTSTRAP_SOURCES})
-target_link_libraries( win32_bootstrap
-					   PRIVATE
-					   oglbase
-					   uibase
-					   shaderunner
-					   )
-
-add_custom_command( TARGET win32_bootstrap POST_BUILD
-					COMMAND ${CMAKE_COMMAND} -E copy_if_different
-							"${PROJECT_SOURCE_DIR}/${LIB_DIR}/glew/bin/Release/x64/glew32.dll"
-							$<TARGET_FILE_DIR:win32_bootstrap>
-							)
-
-install( TARGETS win32_bootstrap
-		 DESTINATION .)
-install( FILES ${LIB_DIR}/glew/bin/Release/x64/glew32.dll
-		 DESTINATION .)
+target_link_libraries(win32_bootstrap
+	PRIVATE
+		oglbase
+		uibase
+		shaderunner
+)
+_common_project_options(win32_bootstrap)
+
+get_target_property(GLEW_DSO_PATH
+	GLEW::GLEW
+	IMPORTED_LOCATION
```

Comment:
This is the property pointing to the DLL file
NOTE: If one day you switch it to a static library you may require to change this portion

Reply:
Please insert the "NOTE" in the code

Reply:
Not sure what you meant by "NOTE" but I added a comment to explain what this call does